transcom/mymove
# Int b 19806 redo

## [B-19806](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19806)

## Summary

This branch adds a couple definitions to the GHC.yaml file that allows users to update the description and expense type fields as part of a PATCH request.


### How to test


1. First, find a move with a moving expense, go to its page as an SC, and use the network dev tools to get the moving expenses ID, eTag, and PPM Shipment ID. (PPM Closeout with locator CLOSE2, from the local dev dataset which is given when running `make_db_e2e_populate` has a moving expense ready that you can use)
2. Using Postman, go to this endpoint: `{{baseUrl}}/ppm-shipments/:ppmShipmentId/moving-expenses/:movingExpenseId` and issue a PATCH request.
3. in the Body, enter an "amount", "description", and "movingExpenseType", the moving expense type must be one of the following strings:

![Screenshot 2024-05-14 at 11 24 50 AM](https://github.com/transcom/mymove/assets/146959895/877b6fe1-aa48-4b48-8fa0-dcca090092ff)

4. Grab the masked_goriila token from the network tab of dev tools and add an "X-CSRF-Token" key under "Headers", with the value of the token.
5. After sending, you should get a response back like the one below, note that the expense type and description have changed.


![Screenshot 2024-05-14 at 10 11 39 AM](https://github.com/transcom/mymove/assets/146959895/db472c18-72e0-4f59-849f-396db5ea533a)
